### PR TITLE
Prevent cylc-run of creating directories for invalid suites

### LINKED ISSUE #3097: Avoid creating empty suite directories when running suites that do not exist
Hi,

I executed `cylc --debug --verbose --no-detach five`, without registering the suite `five`. And I was expecting `cylc` to fail and nothing to be touched on my file system. However, looks like the `~/cylc-run/five` (or wherever is cylc run/work directories specified) was still created, but empty.

This is a bit misleading. I noticed this happened after I moved my cylc run/work directories to `/tmp`, ran my suite `five`, minimized the terminal. Then from another terminal went to check the logs, and the directory was empty.

Took me a few seconds to understand why that happened. The issue happens with `7.8.x` line too, and that's because before checking if the suite exists, we try to create auth files (`create_auth_files` call). I think this logic should be changed, and instead the auth files should be created only if the suite exists.

https://github.com/cylc/cylc/blob/d2e649bfcdaf655a0ae56bf128f213a57e4e612c/lib/cylc/scheduler_cli.py#L94-L104

Cheers
Bruno

### PR DESCRIPTION
**Pull request by kinow**
These changes close #3097

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/scheduler_cli.py [hunk 1]
```diff
@@ -232,13 +232,18 @@ def scheduler_cli(parser, options, args, is_restart=False):
     except SuiteServiceFileError as exc:
         sys.exit(exc)
 
+    suite_srv_dir = SuiteSrvFilesManager().get_suite_srv_dir(reg)
+
+    if not os.path.exists(suite_srv_dir):
+        sys.stderr.write(f'suite service directory not found '
+                         f'at: {suite_srv_dir}\n')
```
Comment (kinow) [inline cylc/flow/scheduler_cli.py]: Shamelessly copied this approach from `cylc-cat-log`.

**Comment by kinow** [comment]
```bash
$ cylc run mp3
suite service directory not found at: /home/kinow/cylc-run/mp3/.service
```

**Comment by hjoliver** [comment]
One test failed, probably real, or else a huge coincidence!
```
./flakytests/registration/02-on-the-fly.t              (Wstat: 0 Tests: 9 Failed: 3)
```

**Comment by hjoliver** [comment]
I predict it'll fail again :grimacing: :
```
$ ./etc/bin/run-functional-tests.sh -v flakytests/registration/02-on-the-fly.t 

ok 1 - 02-on-the-fly-pwd-run
ok 2 - 02-on-the-fly-pwd-run.stdout-contains-ok
ok 3 - 02-on-the-fly-pwd-stop
not ok 4 - 02-on-the-fly-cylc-run-dir-run
not ok 5 - 02-on-the-fly-cylc-run-dir-run.stdout-contains-ok
not ok 6 - 02-on-the-fly-cylc-run-dir-stop
ok 7 - 02-on-the-fly-cylc-run-dir-2-validate
ok 8 - 02-on-the-fly-cylc-run-dir-2-validate.stdout-contains-ok
ok 9 - 02-on-the-fly-cylc-run-dir-2-validate.stderr-contains-ok

    stdout and stderr stored in: /tmp/oliverh/cylctb-20191014T090638Z/registration/02-on-the-fly
Failed 3/9 subtests 
[22:06:53]

Test Summary Report
-------------------
flakytests/registration/02-on-the-fly.t (Wstat: 0 Tests: 9 Failed: 3)
  Failed tests:  4-6
Files=1, Tests=9, 15 wallclock secs ( 0.02 usr  0.00 sys +  2.04 cusr  0.40 csys =  2.46 CPU)
Result: FAIL
```

**Comment by kinow** [comment]
Was re-creating my virtual env to run it, but you beat me to it. Let me take a look at the code.  Thanks!

**Comment by kinow** [comment]
Take 2. The code was checking for `~/cylc-run/blabla/.service`, which prevented the on-the-fly registration. I updated the code to now use `get_suite_run_dir`, and then test for `~/cylc-run/blabla`. 

Wondering if this is good enough, or if we must prevent the `cylc run` of running if:

- not `suite.rc` or 
- no .service dir as before ?

**Comment by hjoliver** [comment]
(I'll have to think about that ... tomorrow)

**Pull request merged**